sycl/tools/sycl-ls/sycl-ls.cpp

```diff
@@ -106,14 +102,26 @@ int main(int argc, char **argv) {
     return EXIT_FAILURE;
   }
 
+  const char *filter = std::getenv("SYCL_DEVICE_FILTER");
+  if (filter) {
```

Comment:
Can we force device filter mechanism to ignore SYCL_DEVICE_FILTER here?

Reply:
As I mentioned above, sycl-ls is a user-level application.
All user application will be affected if the user sets SYCL_DEVICE_FILTER.
'sycl-ls' is designed to show what devices are available in the system and how SYCL_DEVICE_FILTER affects device availability.
Please note that all apps that calls device::get_devices() or any device selectors will be affected by SYCL_DEVICE_FILTER.

Reply:
As soon as SYCL_DEVICE_FILTER can be set in sycl.conf we should warn user at least in the documentation.
In any case current check is insufficient. We can also check sycl.conf using direct file access operation.

Reply:
That's a good point. 
Users can set this env var in sycl.conf file.
The returned devices will be the same because sycl-ls is calling SYCL RT to get devices.
But, sycl-ls will not show the warning message.
I think your original suggestion makes sense now. Using SYCLConfig.

Reply:
Vladimir and I decided not to work on using SYCLConfig because it is an internal SYCL object and it is not possible to use it from the user app.